cql/ManageRareAbnormality.cql

```diff
@@ -1413,91 +1468,108 @@ define RecommendationForHistologyResults:
           'Primary hrHPV testing or cotesting is recommended in 1 year.'
         }
       }
-    when ( // I4.5
+    when ( // I4.4
       AgeInYears() >= 25 and
-      HistologyInterpretedAsCin1OrNormal and
       (
-        PrecedingCytologyIsHsil or
-        PrecedingCytologyIsAscH
-      ) and
-      (
-        not (
-          MostRecentTreatmentDate 1 year or less after Collate.MostRecentBiopsyReport.date
-        ) or
-        MostRecentTreatment is null
-      ) and
+        HistologyInterpretedAsCin1OrNormal and
+        (
+          PrecedingCytologyIsHsil or
+          PrecedingCytologyIsAscH
+        ) and
+        (
+          not (
+            MostRecentTreatmentDate 2 years or less after Collate.MostRecentBiopsyReport.date
+          ) or
+          MostRecentTreatment is null
+        ) and
+        (
+          (
+            (MostRecentCytologyReport is not null) and
+            not CytologyInterpretedAsNilm and
+            not CytologyInterpretedAsHsil and
+            MostRecentCytologyReport.date less than 3.5 years after Collate.MostRecentBiopsyReport.date and
+            MostRecentCytologyReport.date more than 6 months after Collate.MostRecentBiopsyReport.date 
+          ) or
+          (
+            (Collate.MostRecentHpvReport is not null) and
+            Collate.MostRecentHpvResult != 'HPV-negative' and
+            Collate.MostRecentHpvReport.date less than 3.5 years after Collate.MostRecentBiopsyReport.date and
+            Collate.MostRecentHpvReport.date more than 6 months after Collate.MostRecentBiopsyReport.date
+          )
+        )
+      ) or
       (
+        SecondMostRecentHistologyInterpretedAsCin1OrNormal and
         (
-          CytologyInterpretedAsHsil and
-          MostRecentCytologyReport.date 1 year or more after Collate.MostRecentBiopsyReport.date and
-          MostRecentCytologyReport.date less than 3 years after Collate.MostRecentBiopsyReport.date
-        ) or
+          PrecedingCytologyBeforeSecondMostRecentBiopsyIsHsil or
+          PrecedingCytologyBeforeSecondMostRecentBiopsyIsAscH
+        ) and
         (
-          CytologyInterpretedAsAscH and
-          MostRecentCytologyReport.date 2 years or more after Collate.MostRecentBiopsyReport.date and
-          MostRecentCytologyReport.date less than 3 years after Collate.MostRecentBiopsyReport.date
+          not (
+            MostRecentTreatmentDate 2 years or less after SecondMostRecentBiopsyReport.date
+          ) or
+          MostRecentTreatment is null
+        ) and
+        (
+          (SecondMostRecentBiopsyReport is not null) and
+          not SecondMostRecentHistologyInterpretedAsNormal and
```

Comment:
Shouldn't this and the line above it be referring to the MostRecentBiopsyReport? Saying `not SecondMostRecentHistologyInterpretedAsNormal` seems to contradict the line 13 lines above:

>         SecondMostRecentHistologyInterpretedAsCin1OrNormal and

This would be referring to the biopsy taken during the 1st of the annual follow-ups after the original <=CIN1, meaning it would be the most recent. The second most recent would be the original <=CIN1.

Reply:
This should be 'not HistologyInterpretedAsNormal' - will fix

Reply:
I saw that you updated the line I referenced above. Shouldn't `(SecondMostRecentBiopsyReport is not null) and` also be referring to the MostRecent (the line directly above it)? Again, we already know that the SecondMostRecentBiopsyReport is not null because to get to this part of the logic the SecondMostRecentBiopsyResult has to be CIN1 or Normal.